src/tests/JIT/Directed/coverage/importer/Desktop/byrefsubbyref1.il

```diff
@@ -20,20 +20,23 @@
 ldarg 0
 ldarg 1
 sub
+conv.u
 ret
 }
 .method public static native int byrefsubi4(class ctest& V_1, int32 V_2)
 {
 ldarg 0
 ldarg 1
 sub
+conv.u
 ret
 }
 .method public static native int i4subbyref(int32, class ctest& V_2)
 {
 ldarg 0
 ldarg 1
 sub
+conv.u
```

Comment:
`int32 - &` is simply illegal IL by ECMA Table III.2:

![image](https://user-images.githubusercontent.com/10901265/204674352-a46a0fb7-73a8-42f4-bc94-c32b2a53074b.png)

However: why does the test exist, and why did it succeed until now? It may be to validate existing behavior that we can't break.

Reply:
JIT doesn't promise/offer IL validation there are **lots** of ways you can emit a wrong IL and it will be executed just fine, e.g. see this:

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEBLDaAuIUArgHZoAmIA1AD4ACSABAPoCwAUAN6cC+nnegHYmAYSYwE+GKQoBnJgG0AygE850gLYA6AEpl82TTAC6ajTB0B5YACsYYfJx4cmbpvQCcTDQENDYEzYpPgAzABMTACyTAAUAJQeoUyc7kwuaWn0AKxMAIypmW70KIVFZZkY8vhQTABEABZNdRVpVWDaUChModp5pa5FTJrEGK3usE6Dbvwcs5xAA

here I multiply a string literal by 3.14f and return result as an integer.

Reply:
IMO, the JIT should BADCODE on illegal (not unverifiable) IL, otherwise it becomes de-facto supported and later removing that support can break apps.

Reply:
Right, but isn't it already "supported"? I mean if you insert BADCODE here you might brake existing apps that relied on implicit byref -> nuint conversion (presumably, it won't be a rare case). A less common case [broke ninject](https://github.com/dotnet/runtime/issues/67351) and it also was a blocker for many apps with ninject as a dependency to move to a newer .NET (despite the fact IL was very wrong).

What we can also do here is to convert implicit byref->nuint converstion to an explicit one by emitting GT_CAST ourselves (or rather COMMA(ASG(...) routine) where we see it

Reply:
If we just accept that illegal IL `int32 - &` is defined (I'm not even sure what it should be defined as. `native int`? But only on 32-bit platforms?), then why change the test? We should just fix the JIT to not assert, and either insert an explicit cast, or do something else to let it be handled.

Reply:
btw, if we were to change the test (which I still don't believe is the correct course of action), presumably for `i4subbyref` the `conv.u` should come before the `sub` (and thus convert the `&`)

Reply:
I agree, it's better to make jit more resilient to such patterns, jic. As for test we can leave a note that it's not entirely valid IL according to ECMA.

Reply:
`i4subbyref` does look problematic as `int32 - &` is illegal in of itself. Though, that isn't the method that is causing the assertion in the original issue.

Sorry, I was a bit over-zealous with using `conv.u` - the PR now only affects `byrefsubi4` and the `conv.u` comes after the `sub`.